charts/portworx/templates/hooks/pre-delete/delete-storagecluster.yaml

```diff
@@ -0,0 +1,41 @@
+{{- if .Values.deployCluster }}
+{{- $customRegistryURL := .Values.customRegistryURL | default "none" }}
+{{- $registrySecret := .Values.registrySecret | default "none" }}
+{{- $clusterName := .Values.clusterName }}
+
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: kube-system
+  name: px-hook-delete-storagecluster
+  labels:
+    heritage: {{.Release.Service | quote }}
+    release: {{.Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    app.kubernetes.io/managed-by: {{.Release.Service | quote }}
+    app.kubernetes.io/instance: {{.Release.Name | quote }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-weight": "10"
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  backoffLimit: 0
+  template:
+    spec:
+      {{- if not (eq $registrySecret "none") }}
+      imagePullSecrets:
+        - name: {{ $registrySecret }}
+      {{- end }}
+      serviceAccountName: {{ template "px.hookServiceAccount" . }}
+      restartPolicy: Never
+      containers:
+        - name: delete-storagecluster
+          {{- if eq $customRegistryURL "none" }}
+          image: "bitnami/kubectl:{{ template "px.kubectlImageTag" . }}"
+          {{- else}}
+          image: "{{ $customRegistryURL }}/bitnami/kubectl:{{ template "px.kubectlImageTag" . }}"
+          {{- end}}
+          command: ['/bin/sh',
+                    '-c',
+                    'kubectl -n kube-system delete storagecluster {{ $clusterName }}']
```

Comment:
I think the goal is to wait for stc to disappear before deleting operator. storageCluster should be deleted by helm, the problem is operator is deleted hence storageCluster cannot be deleted.

Reply:
By default, the `kubectl delete` command waits for the deletion to complete unless you pass `--no-wait` flag. So, this ensures that the storagecluster is deleted before the chart deletion proceeds. In the chart deletion, it will just delete the operator.

Reply:
> the problem is operator is deleted hence storageCluster cannot be deleted
That is why, we are deleting the StorageCluster in `pre-delete` and once that is completed the chart deletion will delete the operator.

Reply:
nice, forgot the command will block.